### PR TITLE
Codecov: Show UI coverage as subproject and require minimum coverage

### PR DESCRIPTION
**Pull request by shubhank-saxena**

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -4,3 +4,6 @@ coverage:
     project:
       default:
         threshold: 2
+        tests:
```
Comment (armenzg) [inline .codecov.yml]: I think this should be `javascript` and one indentation to the left..

Comment (armenzg) [reply .codecov.yml]: Based on my understanding of this:
https://docs.codecov.io/docs/commit-status#section-excluding-tests-example

**Comment by armenzg** [comment]
Hi @shubhank-saxena there's some other changes in this PR that do not belong.

**Comment by shubhank-saxena** [comment]
I did make sure that rebase happens on the other branch. Let me revert changes and make only the desired one!

**Comment by shubhank-saxena** [comment]
I think the other changes are removed now!

**Comment by armenzg** [comment]
This looks better thanks!
I will test it tomorrow.

**Comment by armenzg** [comment]
I've re-requested a review so I don't foget.

**Comment by armenzg** [comment]
Hi @shubhank-saxena 
This PR is hard to review because I don't know how to test it on codecov if it would what we want it to do.

Looking at the documentation I think this PR needs changing:
https://docs.codecov.io/docs/commit-status#section-splitting-up-projects-example

The example shows:
```
coverage:
  status:
    project:
      users:
        paths:
          - tests/users
          - app/components/user*
      products:
        paths:
          - tests/products
          - app/components/product*
```

I think I incorrectly asked you to move "ui" under "status".
I've pushed a change to the PR I was using for testing and see what happens:
https://github.com/mozilla/treeherder/pull/5994/commits/57c4c3a1209511350fbbfd0e8371114cf1526d7e

Could you please undo the indentation?

**Comment by shubhank-saxena** [comment]
@armenzg done!

**Pull request merged**